modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php

```diff
@@ -119,7 +119,16 @@ protected function addRenderableLocation(array &$build, ContentEntityInterface $
       return;
     }
 
-    CacheableMetadata::createFromObject($venue)->applyTo($build);
+    $cacheability = new CacheableMetadata();
+
+    $access = $venue->access('view', NULL, TRUE);
+    $cacheability->addCacheableDependency($access);
+
+    if (!$access->isAllowed()) {
+      return;
+    }
+
+    $cacheability::createFromObject($venue)->applyTo($build);
```

Comment:
The code here doesn't propagate the caches properly.
`$cacheability::createFromObject($venue)->applyTo($build);` this even loses cacheability information.